### DISCUSSION TITLE
Supporting scoop

### DISCUSSION THREAD

**Discussion by Lemorz56** [nats-io/nats-server#3488] [General]
I was looking at the choco package but found that it seems outdated which led me to find an issue asking for scoop support. I'm not sure if there are any preferred way to run a nats-server, but to be able to install it through scoop would be handy, at least for me.

I took the freedom to create a manifest for this.
The scoop manifest also has the ability to auto-update via github releases without having to edit the manifest file every time you make a release.

I forked scoop's manifest repo and added the nats-server as you can see here: https://github.com/Lemorz56/Main/commit/3ac8e1523e379c54db3fb8dd796637457015848d

I wanted to check with you before making a PR to add this manifest to the official scoop bucket repo😄

[the remark this answers is not in the document]
**Reply by Lemorz56** [reply]
Any update on this? 😄

[the remark this answers is not in the document]
**Reply by scottf** [reply]
@Lemorz56 
* Not sure where there manifest would go into the repo so have to figure that out. 
* Also I see you say it auto tracks github for updates, great, will there be anything else we have to do besides provide the manifest, for instance publish the manifest somewhere scoop is aware of or is it just downloaded by the user?
* Does scoop take into account that our version is in the form `v<major>.<minor>.<patch>` the manifest puts a `v` in from of `$version`. 
* Does scoop understand that the zip file contains a subdirectory?
* Is scoop a standalone windows exe or is it a script or what is it?
* What does scoop use to extract from archives?
Wondering out loud, if we do scoop, then we will have to do choco and then what else? Just more to maintain.

[the remark this answers is not in the document]
**Reply by Lemorz56** [reply]
I understand all your questions and you bring up good points.

> Also I see you say it auto tracks github for updates, great, will there be anything else we have to do besides provide the manifest, for instance publish the manifest somewhere scoop is aware of or is it just downloaded by the user?

The manifest is published once to the a bucket that the scoop maintainer decides, once it is in the bucket, the user just needs to do `scoop install nats-server`.
After that, the manifest is there and the auto-update feature makes sure so that you do not have to maintain that manifest. Once you've made a release on github via github releases like you do today, it will automatically find the newer version. 

> Does scoop take into account that our version is in the form v<major>.<minor>.<patch> the manifest puts a v in from of $version.

Yes it knows the v in front of your version is not a part of the `v<major>.<minor>.<patch>`

> Does scoop understand that the zip file contains a subdirectory?

Yes, you specify in the manifest if there is a sub directory inside the archive.

> Is scoop a standalone windows exe or is it a script or what is it?

Scoop is a command-line installer for windows. So a CLI tool for windows I'd say.

> What does scoop use to extract from archives?

7zip

> Wondering out loud, if we do scoop, then we will have to do choco and then what else? Just more to maintain.

Personally I'd say no not necessarily. I've seen applications that only support either scoop or choco. There is already a choco package for nats-server but it is really outdated. If you would support scoop, than there is at least an option if you don't want to install the exe manually.

I want to also point out that I've created a manifest already, and I also have created a [PR to the scoop repo](https://github.com/ScoopInstaller/Main/pull/3962), but I wanted to bring the discussion here. I just wanted to prepare the work. I have not merged anything, and I will not merge anything without permission to do so. 
For now I've used it locally (you can install a package locally directly from a JSON file manifest) on a few computers. It works as intended and It's been really smooth to be able to update quickly with scoop.

The choice is of course yours, but these are just my answers and my opinions. Hopefully they cleared up some questions at least😄

[the remark this answers is not in the document]
**Reply by scottf** [reply]
Ok I see, it goes in their (scoops main) repo, not ours. I've looked over the manifest you've built, it looks straightforward to me, basically just tells scoop where the binaries live in our github repo. I'll approve in the PR comments.

[the remark this answers is not in the document]
**Reply by scottf** [reply]
@Lemorz56 The only thing I can think of is after it's committed who is allowed to modify it? Just don't want anyone except us and the scoop maintainers. This is probably the case, but just making sure.

[the remark this answers is not in the document]
**Reply by Lemorz56** [reply]
@scottf Thanks :). Yeah ideally it would only be members of the NATS team. I guess it would be up to the scoop maintainer/owner to make sure that is the case. But good that you asked in the PR too.
